src/pointerwrappers.jl

```diff
@@ -43,6 +43,9 @@ end
 # Non-pointer-type fields get wrapped as a normal PointerWrapper
 PointerWrapper(::Type{T}, pointer) where T = PointerWrapper{T}(pointer)
 
+# Allows to convert a PointerWrapper into type `T`, e.g. convert PointerWrapper{Ptr{Nothing}} in PointerWrapper{Int64}
+PointerWrapper(::Type{T}, pw::PointerWrapper) where T = PointerWrapper{T}(pointer(pw))
```

Comment:
I wonder if we really want to allow this free-for-all style anything goes into anything conversion here. Technically, it is not defined what happens in C if you convert one pointer from a datatype to another, except for `void`. This I'd feel slightly more comfortable if we only allow conversion from/to `Cvoid`. Would that be a limitation for what you're trying to achieve?

Reply:
No, I only need a conversion from `Cvoid` to `Int`. So that is fine.

Reply:
Then let’s try to restrict the function signature such that only conversions from or to void are possible.

Reply:
:+1:

Reply:
Done in c6d5ae3a9d2be1b8f200431f9eb33f796850ba70. What I actually need in Trixi.jl is
```julia
PointerWrapper(::Type{T}, pw::PointerWrapper{Ptr{Nothing}}) where T = PointerWrapper{T}(pointer(pw))
```